### DISCUSSION TITLE
How to make the text in a LineEdit comply to specific alignment?

### DISCUSSION THREAD

**Discussion by peterjc123** [slint-ui/slint#2513] [Q&A] [answered]
I tried something below. The preview looks good but it just doesn't work when the program is actually built.
```
LineEdit {
    height: 50px;
    placeholder-text: "Some random text";
    horizontal-alignment: center;
}
```
Platform: ArchLinux, Gnome 43, GDM
Slint version: 1.0.0
Toolchain: Rust 1.68.2

**Comment by tronical** [comment]
Hi @peterjc123 . Can you post a little example that reproduces this?

I tried this in a built program and it looks correct to me:

```
import { LineEdit } from "std-widgets.slint";
export component App inherits Window {
    preferred-width: 300px;
    preferred-height: 200px;
    LineEdit {
        height: 50px;
         placeholder-text: "Some random text";
          horizontal-alignment: center;
    }
}
```

Screenshot:

<img width="335" alt="Screenshot 2023-04-11 at 10 53 18" src="https://user-images.githubusercontent.com/1486/231108438-5348d4c9-cc30-4dfc-b793-15df61fd45d3.png">

For the centering to work, the `LineEdit` needs to have enough width. How do you place your `LineEdit` instance?

**Reply by peterjc123** [reply]
@tronical  Yes, the placeholder text is centered. But if you type something in it, then it keeps left-aligned.

**Reply by tronical** [reply]
You're right! I see it now, too, with the QPainter renderer. This is a bug. I'll convert this into an issue.

**Reply by tronical** [reply, marked as answer]
This is fixed now in https://github.com/slint-ui/slint/pull/2526
